### PR TITLE
D8CORE-2521: Add hook for field permission on open embed field.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW 

# Summary
- Adds field based permissions to embeddable form open embed code field

# Needed By (Date)
- Monday Morning.

# Urgency
- High

# Steps to Test

1. Check out the code in https://github.com/SU-SWS/stanford_media/pull/64
2. Check out this code
3. Run tests in https://github.com/SU-SWS/stanford_profile/pull/273 
or
3. Validate with different users that can or cannot access the field.

### CODE REVIEW

File: stanford_profile_helper.module [hunk 1]
```diff
@@ -603,3 +603,34 @@ function stanford_profile_helper_preprocess_pattern_localfooter(&$variables) {
...
+
+/**
+ * Add a field permission to the open embed code field.
+ *
+ * @param array $form
+ *   The form array.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ */
+function _stanford_profile_helper_embeddable_field_access(array &$form, FormStateInterface $form_state) {
```
Comment (pookmish) [inline stanford_profile_helper.module:629]: This isn't getting called when in the media library from the wysiwyg

**Review by pookmish — CHANGES_REQUESTED** [review]

File: stanford_profile_helper.module (hunk 1 above)
Comment (sherakama) [reply stanford_profile_helper.module:626]: FINE!

**Comment by sherakama** [comment]
revised.

File: stanford_profile_helper.module [hunk 2]
```diff
@@ -29,6 +29,17 @@ function stanford_profile_helper_form_alter(&$form, FormStateInterface $form_sta
     // @link https://www.drupal.org/project/views_bulk_operations/issues/3055770#comment-13116724
     unset($form['header']['views_bulk_operations_bulk_form']['select_all']);
   }
+
+  $embeddable_forms = [
+    'media_library_add_form_embeddable',
+    'media_embeddable_edit_form',
+    'media_embeddable_add_form',
```
Comment (pookmish) [inline stanford_profile_helper.module:36]: i was just thinking, shouldn't field permissions module just do these for us?

Comment (sherakama) [reply stanford_profile_helper.module:36]: Oh possibly. 

Let me check on that.

Comment (sherakama) [reply stanford_profile_helper.module:36]: Yeah, they do.

**Comment by sherakama** [comment]
Revised to just include the wysiwyg form.

**Review by imonroe — APPROVED** [review]
looks straightforward to me.

**Review by pookmish — APPROVED** [review]

**Pull request merged**